gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -737,11 +738,14 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
         // Currently we assume a plan to be transformable by default.
       }
     } catch {
-      case e: UnsupportedOperationException =>
+      case e @ (_: GlutenNotSupportException | _: UnsupportedOperationException) =>
```

Comment:
shall we remove `UnsupportedOperationException` ?

Reply:
I guess `UnsupportedOperationException` can still be thrown unexpectedly for some reasons in user's environment. And by still catching it here, the execution can possibly pass with the help of Gluten's fallback mechanism. So I feel it may be better to temporarily keep it. Thanks!